cmake/CompilerFlags.cmake

```diff
@@ -99,14 +100,16 @@ ELSEIF ( CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"
       ADD_CXX_DEFINITIONS("-Wno-invalid-source-encoding")
     endif()
 
-    # ADDITIONAL DEBUG-MODE-SPECIFIC FLAGS
+    # ADDITIONAL GCC-SPECIFIC FLAGS
     if ( CMAKE_COMPILER_IS_GNUCXX ) # g++
       ADD_CXX_DEBUG_DEFINITIONS("-ffloat-store") # Improve debug run solution stability
       ADD_CXX_DEBUG_DEFINITIONS("-fsignaling-nans") # Disable optimizations that may have concealed NaN behavior
       ADD_CXX_DEBUG_DEFINITIONS("-D_GLIBCXX_DEBUG") # Standard container debug mode (bounds checking, ...)
+      ADD_CXX_RELEASE_DEFINITIONS("-finline-limit=2000") # More aggressive inlining
     endif()
 
   ADD_CXX_DEBUG_DEFINITIONS("-ggdb") # Produces debugging information specifically for gdb
+  ADD_CXX_RELEASE_DEFINITIONS("-Ofast -fno-stack-protector") # -Ofast (or -ffast-math) needed to auto-vectorize floating point loops
```

Comment:
@mbadams5 Yes but in order to allow auto-vectorization of floating point loops, which is very important for performance, you need to enable -ffast-math. The impact on floating point precision is generally not meaningful (although the changes can be significant with less stable computations) and is similar to changing division to multiplication by the inverse. But, yes, we should discuss all of this. We might want to make special vectorization friendly, high-performance builds and leave a vanilla build as the default.